cmd/kubeapps-apis/plugins/pkg/paginate/paginate.go

```diff
@@ -41,9 +41,16 @@ func PageOffsetFromInstalledRequest(request *corev1.GetInstalledPackageSummaries
 func PageOffsetFromAvailableRequest(request *corev1.GetAvailablePackageSummariesRequest) (int, error) {
 	offset, err := PageOffsetFromPageToken(request.GetPaginationOptions().GetPageToken())
 	if err != nil {
-		return 0, status.Errorf(codes.InvalidArgument, "unable to intepret page token %q: %v",
-			request.GetPaginationOptions().GetPageToken(), err)
+		return 0, err
 	} else {
 		return offset, nil
 	}
 }
+
```

Comment:
Note: it also wasn't obvious to me why we have the two separate functions above that do exactly the same thing, just for different types of requests. We can instead just have call-sites that pass in the pagination options rather than the request, I'd think? (which is what I'm doing below with the similar ItemOffset helper). Anyway, the above will be removed once the other two plugins are similarly updated.

Reply:
Yep, maybe there is a reason for that, but indeed makes more sense having a single function with the `request.GetPaginationOptions()` type as param.

Reply:
ooor, we can start using [generics](https://go.dev/doc/tutorial/generics) :P 
Sometimes I wish go supported overloading...